### PR TITLE
Conversation: add block example

### PR DESCRIPTION
**Pull request by retrofox**
_This PR is part of the [follow-up tasks](https://github.com/Automattic/jetpack/issues/18123) after merging `conversation`/`dialogue` blocks._

This PR defines an example to the conversation block. It will be consumed by the block styles, block preview in the inserter, and other places where the block is previewed.

Inserter | Block styles
--------|----------
image | image

Fixes #

#### Changes proposed in this Pull Request:

* Conversation: add block example

#### Jetpack product discussion

p1610110339335100-slack-

#### Testing instructions:

* Search the `Conversation` block in the inserter. Confirm you see the preview defined by the example.
* Editing a `Conversation` block, open the block settings sidebar, and confirm the preview of styles are defined by the example

#### Proposed changelog entry for your changes:

n/a

### CODE REVIEW

**Review by allilevine — APPROVED** [review]
✅ I see the example in both the inserter and the sidebar:
Screen Shot 2021-01-11 at 2 23 07 PM
Screen Shot 2021-01-11 at 2 23 52 PM

~The example can disappear depending on theme colors:~ I think the iframe content was just taking a while to load.
Screen Shot 2021-01-11 at 2 56 26 PM

**Review by jeherve — CHANGES_REQUESTED** [review]
Thanks for adding a block preview!
Do you think we could have this be in English, with translatable strings, so more site owners can relate to the block?

**Comment by retrofox** [comment]
> Thanks for adding a block preview!
> Do you think we could have this be in English, with translatable strings, so more site owners can relate to the block?

Yes, provably. Let me ask again about this :-)

**Comment by sdixon194** [comment]
Thanks for adding the preview! Some things I noticed while testing:

Seems the height of the columns section in the block settings varies depending on the theme. Likely non-blocking, but couldn't reproduce with other block previews:

**Twenty Twenty One:**

Screen Shot on 2021-01-13 at 12:32:23


**Twenty Twenty:**

Screen Shot 2021-01-13 at 12 30 07


**Twenty Nineteen:** 

Screen Shot 2021-01-13 at 12 28 46


I also couldn't get the block to load at all when running wp 5.5: 

Screen Shot on 2021-01-13 at 11:53:48

@jeherve would you mind taking another look as well? Thanks!

**Comment by retrofox** [comment]
> Thanks for adding the preview! Some things I noticed while testing:
> 
> Seems the height of the columns section in the block settings varies depending on the theme. Likely non-blocking, but couldn't reproduce with other block previews:

The block styles component is implemented by the core API. There isn't too much to do from the Jetpack plugin. We didn't add any specific styles, so I'm afraid it would happen in other blocks too.


> I also couldn't get the block to load at all when running wp 5.5:

Would you mind pasting the error in case it exists from the dev console?

> @jeherve would you mind taking another look as well? Thanks!

**Comment by sdixon194** [comment]
>The block styles component is implemented by the core API. 

Ah yup, figured it was something like that. 

>Would you mind pasting the error in case it exists from the dev console?

Yup! Apologies, should have included that in my initial comment:

```
TypeError: Object(...) is not a function
    ConversationEdit webpack:///./extensions/blocks/conversation/edit.js?:52
```

**Comment by retrofox** [comment]
Thanks!
> ```
> TypeError: Object(...) is not a function
>     ConversationEdit webpack:///./extensions/blocks/conversation/edit.js?:52
> ```

it's werid. Going to take a look again, Could you test the master branch, please?

**Comment by sdixon194** [comment]
>it's werid. Going to take a look again, Could you test the master branch, please?

Yup! Got the same issue with the block not loading, although the preview at least looked better. On 5.5, I just got a generic "this block can not be previewed" message.

Screen Shot on 2021-01-13 at 13:58:05

**Comment by retrofox** [comment]
> > Thanks for adding a block preview!
> > Do you think we could have this be in English, with translatable strings, so more site owners can relate to the block?
> 
> Yes, provably. Let me ask again about this :-)

cc @sfougnier

**Comment by retrofox** [comment]
> I also couldn't get the block to load at all when running wp 5.5:

Should we ignore this? since we support 5.6 or later if I'm not wrong. cc @jeherve

**Comment by jeherve** [comment]
> since we support 5.6 or later if I'm not wrong

We'll need to support WordPress 5.5 until WP 5.7 is released, so that's something we'll need to fix before the launch of that block. We can address that in a follow-up PR though.

**Comment by retrofox** [comment]
> > since we support 5.6 or later if I'm not wrong
> 
> We'll need to support WordPress 5.5 until WP 5.7 is released, so that's something we'll need to fix before the launch of that block. We can address that in a follow-up PR though.

[High] priority btw

**Comment by sfougnier** [comment]
> Thanks for adding a block preview!
Do you think we could have this be in English, with translatable strings, so more site owners can relate to the block?

There was a discussion on Slack yesterday regarding the block preview content that I wanted to follow up and document here.

Judging based on content from other blocks, it seems to be a mix of quotes from Matt and references to classic literature. Maybe we could try pulling from some Shakespeare text since it's public domain and correlates with the purpose of the block?

I'm not really Shakespeare-savvy so feel free to call out an alternative, but I pulled this:

**ROSALIND**
O, my dear Orlando, how it grieves me to see thee wear thy heart in a scarf!
**ORLANDO**
It is my arm.
**ROSALIND**
I thought thy heart had been wounded with the claws of a lion.
**ORLANDO**
Wounded it is, but with the eyes of a lady.

**Review by jeherve — CHANGES_REQUESTED** [review]
This looks good to me. Should we make all this translatable, so folks can see the conversation in their own language in their dashboard?

**Comment by retrofox** [comment]
image

File: projects/plugins/jetpack/extensions/blocks/conversation/example.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+		attributes: {
+			content: 'Shakespeare text',
+			level: 4,
+		},
+	},
+	{
+		name: 'jetpack/dialogue',
+		attributes: {
+			...participants[ 0 ],
+			content: 'O, my dear Orlando, how it grieves me to see thee wear thy heart in a scarf!',
```
Comment (jeherve) [inline projects/plugins/jetpack/extensions/blocks/conversation/example.js:28]: To clarify, I meant making this translatable, e.g.

```suggestion
			content: __( 'O, my dear Orlando, how it grieves me to see thee wear thy heart in a scarf!', 'jetpack' ),
```

Comment (retrofox) [reply projects/plugins/jetpack/extensions/blocks/conversation/example.js]: sorry commeted 6343cb34011ee2b8b450e2e36dc754eca63274c0 before to ready your suggestion :-(

File: projects/plugins/jetpack/extensions/blocks/conversation/example.js [hunk 2]
```diff
@@ -0,0 +1,83 @@
+/**
+ * WordPress dependencies
+ */
+
+ import { __ } from '@wordpress/i18n';
+
+const participants = [
+	{
+		participantSlug: 'participant-0',
+		participant: 'Rosalind',
```
Comment (jeherve) [inline projects/plugins/jetpack/extensions/blocks/conversation/example.js:10]:
```suggestion
		participant: __( 'Rosalind', 'jetpack' ),
```

File: projects/plugins/jetpack/extensions/blocks/conversation/example.js [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+		hasBoldStyle: true,
+		hasUppercaseStyle: true,
+	},
+	{
+		participantSlug: 'participant-1',
+		participant: 'Orlando',
```
Comment (jeherve) [inline projects/plugins/jetpack/extensions/blocks/conversation/example.js:16]:
```suggestion
		participant: __( 'Orlando', 'jetpack' ),
```

File: projects/plugins/jetpack/extensions/blocks/conversation/example.js [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+		hasItalicStyle: true,
+		hasUppercaseStyle: true,
+	},
+];
+
+const template = [
+	{
+		name: 'core/heading',
+		attributes: {
+			content: 'Shakespeare text',
```
Comment (jeherve) [inline projects/plugins/jetpack/extensions/blocks/conversation/example.js:26]:
```suggestion
			content: __( 'Shakespeare text', 'jetpack' ),
```

**Review by jeherve — CHANGES_REQUESTED** [review]

File: projects/plugins/jetpack/extensions/blocks/conversation/example.js (hunk 2 above)
Comment (retrofox) [reply projects/plugins/jetpack/extensions/blocks/conversation/example.js:10]: I think we shouldn't translate the names.

File: projects/plugins/jetpack/extensions/blocks/conversation/example.js (hunk 3 above)
Comment (retrofox) [reply projects/plugins/jetpack/extensions/blocks/conversation/example.js:16]: same. I think we shouldn't translate the names.

**Review by jeherve — APPROVED** [review]

**Comment by retrofox** [comment]
r219501-wpcom

**Pull request merged**